Decrease margins around the editor and project manager windows

Calinou opened the pull request:
This increases the screen real estate available for Godot, especially at smaller window sizes.

## Preview

***Edit:** This preview is outdated; there's now 1 pixel of margin on all sides (instead of 0).*

### Editor - Before

![Before](https://user-images.githubusercontent.com/180032/87227127-17818c00-c399-11ea-8cc6-9ba6c7596264.png)

### Editor - After

![image](https://user-images.githubusercontent.com/180032/87431798-6de40a00-c5e7-11ea-8434-6ef24f7ffe09.png)

### Project manager - Before

![Before](https://user-images.githubusercontent.com/180032/87227134-1fd9c700-c399-11ea-803c-3e4022eeb0a8.png)

### Project manager - After

![After](https://user-images.githubusercontent.com/180032/87227135-20725d80-c399-11ea-82be-03ead1882128.png)

Calinou committed:
Decrease margins around the editor and project manager windows

This increases the screen real estate available for Godot, especially
at smaller window sizes.

The editor icons for tab options were resized from 6×16 to 16×16
so they don't touch the window edge. Incindentally, this also makes them
easier to click.

Ansraer commented:
Here is a before and after slider: https://imgsli.com/MTg5MzM

YeldhamDev commented:
Honestly? I think it looks ugly. Having no margin borders doesn't really fit the current design of the editor.

Calinou commented:
@YeldhamDev It's normal for it to look out of place at first, but I think you get used to it after an hour of usage or so. It looked weire to me as well when I decreased it at first. We're just so used to those 8-pixel margins we don't even pay attention to them anymore.

Also, keep in mind the real estate gains are significant on low-resolution displays like 1366x768 (which is common on budget laptops in use today). I've seen lots of people using Godot on such small displays over the years, and it looks like it will be the case for the foreseeable future. Let's make their experience as good as possible :slightly_smiling_face:

YeldhamDev commented:
> Also, keep in mind the real estate gains are significant on low-resolution displays like 1366x768 [...]

That happens to be the size of my monitor! :wink: And I think those margins make very little difference. I would personally wait for the new editor theme in 4.0 to do this kind of change, or at the very least, resize them down to 4 instead of completely removing them.

aaronfranke commented:
I like it much better with the margins, personally. I think this is mostly due to the fact that modern window managers don't have borders on the sides and bottom. I don't really buy the low-res screens argument, for one, I think it makes sense for Godot to target 1080p screens, and for another, 16 pixels is only a 2% height difference on a 1366x768 screen anyway.

> The editor icons for tab options were resized from 6×16 to 16×16 so they don't touch the window edge. Incindentally, this also makes them easier to click.

This should be done regardless of whether or not we get rid of margins, sounds like a nice improvement :)

YeldhamDev commented:
@aaronfranke
>[The icon resize] should be done regardless of whether or not we get rid of margins, sounds like a nice improvement :)

Already present in a PR of mine: #40289

hilfazer commented:
No! Let's leave efficient space usage to competing game engines. They enjoy having advantages over Godot.

Just kidding, great work.

Calinou commented:
@hilfazer That's another of my arguments for making margins as small as we can. None of the other tools in the open source creative world (Blender, Krita, …) have significant window margins like we do.

groud commented:
Hmm, I think this is the way we have to go, but as is, it kind of looks weird. 
If we go for a more compact look we should also decrease the internal margins (at least the ones between the docks and the main area), otherwise it kind of creates an imbalanced look.

Also, the "3 dots button" at the top of the right dock needs to be slightly moved to the left.

Calinou commented:
> If we go for a more compact look we should also decrease the internal margins (at least the ones between the docks and the main area), otherwise it kind of creates an imbalanced look.

I tried to do this but I didn't get it to work :slightly_frowning_face: 
This is why I'd prefer keeping that for a later PR.

YeldhamDev commented:
@Calinou The images should be updated.

Calinou commented:
@YeldhamDev It looks identical, I just rebased to use your icons instead of mine.

bruvzg commented:
On macOS (and probably some X11 display managers) few pixels in the corners are cut and blurred by window border, with the zero margins it can cut into controls.
![Screenshot 2020-07-14 at 16 59 55](https://user-images.githubusercontent.com/7645683/87435676-b9e97b80-c5f4-11ea-93c0-6307f148ef08.png)

Calinou commented:
@bruvzg I think there's still enough internal margin to make this a non-issue. (Text and icons never really *touch* the window's edges with the margins removed.)

akien-mga commented:
Needs a rebase to fix CI.

I'm not 100% convinced by this change either. At least the provided screenshots don't show a significant increase in available usable space, and it's now less consistent not to have padding near the window borders while there is still significant padding between elements. But I'm not opposed to it either.

name-here commented:
Only decreasing the outer margins seems like an odd choice, since, as mentioned by others, it makes the editor feel unbalanced.  I think decreasing the internal margins would be an improvement, though, and decreasing the external margins might go well with that.  For reference, since *everything* in Godot has a few pixels margin on all sides, its margins between usable spaces are around 4x as big as Blender's.

<img width="150" alt="Screen Shot 2020-11-29 at 4 03 15 PM" src="https://user-images.githubusercontent.com/26884090/100553760-b1b97200-325d-11eb-8a9b-8b5c52e562f3.png">
<img width="150" alt="Screen Shot 2020-11-29 at 4 03 53 PM" src="https://user-images.githubusercontent.com/26884090/100553761-b1b97200-325d-11eb-9805-161e9bbbd167.png">


Personally, I think Godot could use some larger changes to it's layout, too.  For example, why are 2D, 3D (GD)Script and AssetLib "special" workspaces, while the shader editors aren't?  Why can't I have, say, the 2D and Script editors open at the same time, but I can have 2D and the shader editor open together?  Why can't 2 scripts be open side by side?  I like Godot's layout, but I wish it had the flexibility other tools have.

aaronfranke commented:
> @name-here: For example, why are 2D, 3D (GD)Script and AssetLib "special" workspaces, while the shader editors aren't?

The idea is that usually you want to visually see the result of the shader while you're working on it, so you want to have either the 2D or 3D main screens visible. Maybe this can be changed for 4.x when we get the ability to pop out things to their own windows.

Calinou commented:
Rebased and tested again, it works successfully.

I changed it so there's now one pixel of margin on all sides. This prevents issues with some displays partially occluding the borders of controls or text due to their geometry.

Zireael07 commented:
Why close?

Calinou commented:
> Why close?

There's no consensus on the implementation, as the margins look uneven. We'd need SplitContainer to be able to have a virtual dragging area before we can reduce its margins, making the editor margins consistent.

davthedev commented:
Here is another approach I just worked on: expose the values as theme constants so that anyone can make a custom theme and decrease the margins through it.

https://github.com/godotengine/godot-proposals/issues/6482
PR: https://github.com/godotengine/godot/pull/74767

> We'd need [SplitContainer to be able to have a virtual dragging area](https://github.com/godotengine/godot-proposals/issues/3635) 

This is now supported: https://github.com/godotengine/godot/pull/65355
